mayastor/src/bdev/uring_util.rs

```diff
@@ -0,0 +1,89 @@
+use std::{
+    fs::{File, OpenOptions},
+    io::{BufRead, BufReader, ErrorKind},
+    os::unix::fs::OpenOptionsExt,
+};
+
+pub fn fs_supports_direct_io(path: &str) -> bool {
+    // SPDK uring bdev uses IORING_SETUP_IOPOLL which is usable only on a file
+    // descriptor opened with O_DIRECT. The file system or block device must
+    // also support polling.
+    // This works on at least XFS filesystems
+    match OpenOptions::new()
+        .read(true)
+        .write(true)
+        .custom_flags(libc::O_DIRECT)
+        .open(path)
+    {
+        Ok(_f) => true,
+        Err(e) => {
+            assert_eq!(e.kind(), ErrorKind::InvalidInput);
+            println!("Skipping uring bdev, open: {:?}", e);
+            false
+        }
+    }
+}
+
+fn get_mount_filesystem(path: &str) -> Option<String> {
```

Comment:
just a remark: we have a bunch of mount related code in csi/src/mount.rs that can serve as an inspiration. But if this code works ok, then I guess it's fine. Sharing the code between ms and agent would be too difficult.